app/controllers/hub/organizations_controller.rb

```diff
@@ -21,7 +21,9 @@ def create
     end
 
     def show
-      @organization = VitaPartner.organizations.find(params[:id])
+      @organization = @vita_partner
+      raise ActionController::RoutingError.new('Not Found') unless @vita_partner.organization?
```

Comment:
Still raises an error when you can't find the organization _without_ doing a second database lookup for the organization.

Reply:
Hmm... I actually disagree for the reason that we did already _find_ the record -- we just aren't allowing you to route to it because it's not an organization and thus does not fulfill the routing requirements for this controller method. I'm open to changing the order of the variable definitions, and I actually think it's not necessary at all to use @organization instead of @vita_partner in the template. To me, the main goal was to stop doing another find for a record we already had available to us.